orttraining/orttraining/training_ops/cpu/nn/broadcast_grad_args.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/training_ops/cpu/nn/broadcast_grad_args.h"
+
+namespace onnxruntime {
+namespace contrib {
+#define REGISTER_KERNEL_TYPED(T)                                  \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
+      BroadcastGradientArgs,                                      \
+      kMSDomain,                                                  \
+      1,                                                          \
+      T,                                                          \
+      kCpuExecutionProvider,                                      \
+      KernelDefBuilder()                                          \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      BroadcastGradientArgs<T>);
+
+REGISTER_KERNEL_TYPED(int64_t)
+
+template <typename T>
```

Comment:
does this need to be templated? is T always the index type, int64_t?

Reply:
The TensorFlow kernel also supports type int32 as well, this allows us to add that support easily in the future

Reply:
we can use something like this to avoid templating the class:
https://github.com/microsoft/onnxruntime/blob/5086e55a35f83e3137bdb34b6d7210c97a512e6a/include/onnxruntime/core/framework/data_types_internal.h#L300
usage example: https://github.com/microsoft/onnxruntime/blob/7627e6bcc26893a96344ad90fabb11cd76530348/onnxruntime/core/providers/cpu/math/clip.cc#L73
i think reducing the number of classes can reduce binary size.